cpp/src/structs/utilities.cpp

```diff
@@ -98,11 +98,20 @@ struct table_flattener {
   std::vector<null_order> flat_null_precedence;
   column_nullability nullability;
 
+  rmm::cuda_stream_view stream;
+  rmm::mr::device_memory_resource* mr;
+
   table_flattener(table_view const& input,
                   std::vector<order> const& column_order,
                   std::vector<null_order> const& null_precedence,
-                  column_nullability nullability)
-    : column_order(column_order), null_precedence(null_precedence), nullability(nullability)
+                  column_nullability nullability,
+                  rmm::cuda_stream_view stream,
+                  rmm::mr::device_memory_resource* mr)
+    : column_order(column_order),
+      null_precedence(null_precedence),
+      nullability(nullability),
+      stream(stream),
+      mr(mr)
```

Comment:
Since we are already touching these lines: prefer `{}` initializer syntax. It does not allow narrowing conversions and avoids ambiguity.
[ES.23: Prefer the {}-initializer syntax](https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#es23-prefer-the--initializer-syntax
).
Also: [What are the advantages of list initialization (using curly braces)?](https://stackoverflow.com/questions/18222926/what-are-the-advantages-of-list-initialization-using-curly-braces)

```suggestion
    : column_order{column_order},
      null_precedence{null_precedence},
      nullability{nullability},
      stream{stream},
      mr{mr}
```